app/javascript/widget/composables/useAvailability.js

```diff
@@ -0,0 +1,131 @@
+import { computed } from 'vue';
+import { utcToZonedTime } from 'date-fns-tz';
+import { isTimeAfter } from 'shared/helpers/DateHelper';
+import { useI18n } from 'dashboard/composables/useI18n';
+import { useNextAvailability } from './useNextAvailability';
+/**
+ * Composable for handling availability-related computations.
+ * @param {Object} t - Vue i18n instance for translations.
+ * @returns {Object} An object containing computed properties and methods for availability management.
+ */
+export function useAvailability(availableAgents) {
+  const { timeLeftToBackInOnline } = useNextAvailability();
+  const { t } = useI18n();
+  const channelConfig = computed(() => window.chatwootWebChannel);
+  const replyTime = computed(() => window.chatwootWebChannel.replyTime);
+
+  const replyTimeStatus = computed(() => {
+    switch (replyTime.value) {
+      case 'in_a_few_minutes':
+        return t('REPLY_TIME.IN_A_FEW_MINUTES');
+      case 'in_a_few_hours':
+        return t('REPLY_TIME.IN_A_FEW_HOURS');
+      case 'in_a_day':
+        return t('REPLY_TIME.IN_A_DAY');
+      default:
+        return t('REPLY_TIME.IN_A_FEW_HOURS');
+    }
+  });
+
+  const replyWaitMessage = computed(() => {
+    const { workingHoursEnabled } = channelConfig.value;
+    if (workingHoursEnabled) {
+      return isOnline.value
```

Comment:
There is some Eslint issue with `isOnline` computed property `'isOnline' was used before it was defined.`. Please take a look.